src/xpk/core/storage.py

```diff
@@ -0,0 +1,365 @@
+      xpk_exit(1)
+  return storages
+
+
+def get_storages_to_mount(
+    k8s_api_client: ApiClient, requested_storages: list[str]
+) -> list[Storage]:
+  """
+  Retrieves a list of Storage resources by their names, including auto-mounted storages.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+      names: A list of Storage resource names to retrieve.
+
+  Returns:
+      A list of Storage objects matching the given names and any auto-mounted storages.
+  """
+  storages = get_storages(k8s_api_client, requested_storages)
+  for auto_mounted_stg in get_auto_mount_storages(k8s_api_client):
+    # prevent duplicating storages
+    if auto_mounted_stg.name not in requested_storages:
+      storages.append(auto_mounted_stg)
+
+  return storages
+
+
+def get_storage(k8s_api_client: ApiClient, name: str) -> Storage:
+  """
+  Retrieves a specific Storage custom resource by its name.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+      name: The name of the Storage resource to retrieve.
+
+  Returns:
+      A Storage object representing the retrieved Storage resource.
+  """
+  api_instance = k8s_client.CustomObjectsApi(k8s_api_client)
+  try:
+    resp = api_instance.get_cluster_custom_object(
+        name=name,
+        group=XPK_API_GROUP_NAME,
+        version=XPK_API_GROUP_VERSION,
+        plural=STORAGE_CRD_KIND.lower() + "s",
+    )
+    return Storage(resp)
+  except ApiException as e:
+    xpk_print(f"Kubernetes API exception while getting Storage {name}: {e}")
+    xpk_exit(1)
+
+
+def install_storage_crd(k8s_api_client: ApiClient) -> None:
+  """
+  Installs the Storage custom resource definition (CRD) in the Kubernetes cluster.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+  """
+  xpk_print(f"Creating a new CRD: {STORAGE_CRD_NAME}")
+  try:
+    utils.create_from_yaml(
+        k8s_api_client,
+        f"{os.path.dirname(__file__)}{STORAGE_CRD_PATH}",
+        verbose=True,
+    )
+    xpk_print(f"Created a CRD: {STORAGE_CRD_NAME} successfully")
+  except FailToCreateError as e:
+    for api_exception in e.api_exceptions:
+      if api_exception.status == 409:
+        xpk_print(
+            f"CRD: {STORAGE_CRD_NAME} already exists. Skipping its creation"
+        )
+      break
+    else:
+      xpk_print(f"Encountered error during installing Storage CRD: {e}")
+      xpk_exit(1)
+
+
+def print_storages_for_cluster(storages: list[Storage], cluster: str):
+  """
+  Prints in human readable manner a table of Storage resources that belong to the specified cluster.
+
+  Args:
+      storages: A list of Storage objects.
+      cluster: The name of the cluster to filter by.
+  """
+  headers = [
+      "NAME",
+      "TYPE",
+      "AUTO MOUNT",
+      "MOUNT POINT",
+      "READONLY",
+      "MANIFEST",
+  ]
+  storage_tab = []
+  for storage in storages:
+    if storage.cluster == cluster:
```

Comment:
How can we have storages for different clusters here?

Reply:
That's a valid point, thanks, done